server/container_remove.go

```diff
@@ -22,11 +21,8 @@ func (s *Server) RemoveContainer(ctx context.Context, req *pb.RemoveContainerReq
 		return nil, fmt.Errorf("failed to update container state: %v", err)
 	}
 
-	cState := s.runtime.ContainerStatus(c)
-	if cState.Status == oci.ContainerStateCreated || cState.Status == oci.ContainerStateRunning {
-		if err := s.runtime.StopContainer(c); err != nil {
-			return nil, fmt.Errorf("failed to stop container %s: %v", c.ID(), err)
-		}
+	if err := s.runtime.StopContainer(c); err != nil {
```

Comment:
I think it may be better to retain state check outside rather than move it into StopContainer.